aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -161,13 +164,160 @@ func processingConfigurationSchema() *schema.Schema {
+				"cloudwatch_logging_options": flattenCloudwatchLoggingOptions(*destination.ElasticsearchDestinationDescription.CloudWatchLoggingOptions),
+			}
+			elasticsearchConfList := make([]map[string]interface{}, 1)
+			elasticsearchConfList[0] = elasticsearchConfiguration
+			d.Set("elasticsearch_configuration", elasticsearchConfList)
+			d.Set("s3_configuration", flattenFirehoseS3Configuration(*destination.ElasticsearchDestinationDescription.S3DestinationDescription))
+		} else if destination.S3DestinationDescription != nil {
+			d.Set("destination", "s3")
+			d.Set("s3_configuration", flattenFirehoseS3Configuration(*destination.S3DestinationDescription))
+		} else if destination.ExtendedS3DestinationDescription != nil {
+			d.Set("destination", "extended_s3")
+
+			extendedS3Configuration := map[string]interface{}{
+				"buffering_interval":         *destination.ExtendedS3DestinationDescription.BufferingHints.IntervalInSeconds,
+				"buffering_size":             *destination.ExtendedS3DestinationDescription.BufferingHints.SizeInMBs,
+				"bucket_arn":                 *destination.ExtendedS3DestinationDescription.BucketARN,
+				"role_arn":                   *destination.ExtendedS3DestinationDescription.RoleARN,
+				"compression_format":         *destination.ExtendedS3DestinationDescription.CompressionFormat,
+				"prefix":                     *destination.ExtendedS3DestinationDescription.Prefix,
+				"s3_backup_mode":             *destination.ExtendedS3DestinationDescription.S3BackupMode,
+				"cloudwatch_logging_options": flattenCloudwatchLoggingOptions(*destination.ExtendedS3DestinationDescription.CloudWatchLoggingOptions),
+			}
+			if destination.ExtendedS3DestinationDescription.EncryptionConfiguration.KMSEncryptionConfig != nil {
+				extendedS3Configuration["kms_key_arn"] = *destination.ExtendedS3DestinationDescription.EncryptionConfiguration.KMSEncryptionConfig
+			}
+			if destination.ExtendedS3DestinationDescription.ProcessingConfiguration != nil {
+				extendedS3Configuration["processing_configuration"] = flattenProcessingConfiguration(*destination.ExtendedS3DestinationDescription.ProcessingConfiguration)
+			}
+			extendedS3ConfList := make([]map[string]interface{}, 1)
+			extendedS3ConfList[0] = extendedS3Configuration
+			d.Set("extended_s3_configuration", extendedS3ConfList)
+		}
+		d.Set("destination_id", *destination.DestinationId)
+	}
+	return nil
+}
+
 func resourceAwsKinesisFirehoseDeliveryStream() *schema.Resource {
 	return &schema.Resource{
 		Create: resourceAwsKinesisFirehoseDeliveryStreamCreate,
 		Read:   resourceAwsKinesisFirehoseDeliveryStreamRead,
 		Update: resourceAwsKinesisFirehoseDeliveryStreamUpdate,
 		Delete: resourceAwsKinesisFirehoseDeliveryStreamDelete,
 
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				resARN, err := arn.Parse(d.Id())
+				if err != nil {
+					return []*schema.ResourceData{}, err
```

Comment:
@radeksimko Is this okay or should it be `return nil, err` instead?

Reply:
It likely doesn't matter if a non-nil error is returned - so returning `nil` here is nicer.